core/src/components/checkbox/checkbox.tsx

```diff
@@ -98,6 +98,22 @@ export class Checkbox implements ComponentInterface {
    */
   @Prop() alignment: 'start' | 'center' = 'center';
 
+  /**
+   * If `true`, the checkbox will be presented with an error style when it is unchecked.
+   */
+  @Prop() required = false;
+
+  /**
+   * Set to `"soft"` for a checkbox with more rounded corners.
+   */
+  @Prop({ reflect: true }) shape?: 'soft' | 'rectangular' = 'soft';
```

Comment:
I would recommend removing the `{ reflect: true }` for now. This applies to both usages. 

This is some history here, but Ionic Framework currently uses reflection of CSS classes on the element host, to use for the styling selectors. We should eventually move away from this, as reflecting the attributes is actually more compliant in SSR environments (mark-up provided by the developer matches the rendered output).

Since we are not using these attributes in our current style implementation, we are invoking more JS in Stencil's virtual dom to reflect the attributes to the element.